src/datasets/builder.py

```diff
@@ -276,6 +281,12 @@ class DatasetBuilder:
     # Optional default config name to be used when name is None
     DEFAULT_CONFIG_NAME = None
 
+    # Default batch size used by the ArrowWriter
+    # It defines the number of samples that are kept in memory before writing them
+    # and also the length of the arrow chunks
```

Comment:
From looking at the source, only the `"the length of the arrow chunks"` part seems to be true for the `ArrowWriter` due to using `write_table` there, which always writes to disk (with `max_chunksize=writer_batch_size`)

Reply:
I naively copy pasted the comment from GeneratorBasedBuidler.

What do you think of 
```python
    # Default batch size used by the ArrowWriter.
    # It defines the maximum size of the arrow record batches (or parquet row groups).
    # Setting it to a low value may also help reducing RAM usage in certain cases
    # by flushing the data on disk more frequently.
```

Reply:
Yes, either this or we could check the table's `num_rows` in `write_table` before passing it to the `pa_writer`.

Reply:
I'll think I'd keep it as it is right now to avoid changing the logic (out of scope of this PR).

Lmk if it's fine for you

Reply:
Yes, that's fine